direct/core/src/main/java/cz/o2/proxima/direct/time/WatermarkConfiguration.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.time;
+
+import cz.o2.proxima.time.WatermarkEstimatorFactory;
+import cz.o2.proxima.time.WatermarkIdlePolicyFactory;
+import cz.o2.proxima.util.Classpath;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/** Watermark configuration */
+@Slf4j
+public abstract class WatermarkConfiguration implements Serializable {
+  private static final long serialVersionUID = 1L;
+
+  public static final String CFG_PREFIX = "watermark.";
+
+  @Getter private final Map<String, Object> cfg;
+  @Getter private WatermarkIdlePolicyFactory watermarkIdlePolicyFactory;
+  @Getter private WatermarkEstimatorFactory watermarkEstimatorFactory;
+
+  protected WatermarkConfiguration(Map<String, Object> cfg) {
+    this.cfg = cfg;
+  }
+
+  /** Returns configuration key with added watermark config prefix */
+  public static String cfgKey(String cfgName) {
```

Comment:
the method name is confusing, what about `WatermarkConfiguration#prefixedKey(String)`?

Reply:
Why not, actually it was my na for it in first iteration.